File: Website TSTH2/app/Models/BarangModel.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class BarangModel extends Model
{
    use HasFactory;
    protected $table = "tbl_barang";
    protected $primaryKey = 'barang_id';
    protected $fillable = [
        'jenisbarang_id',
        'satuan_id',
        'gudang_id',
        'barang_kode',
        'barang_nama',
        'barang_slug',
        'barang_harga',
        'barang_stok',
        'barang_gambar',
        'jenis_barang',
        'barcode'
    ];
    public function jenisBarang()
    {
        return $this->belongsTo(JenisBarangModel::class, 'jenisbarang_id', 'jenisbarang_id');
    }
    public function satuan()
{
    return $this->belongsTo(SatuanModel::class, 'satuan_id', 'satuan_id');
}
public function gudang()
{
    return $this->belongsTo(GudangModel::class, 'gudang_id', 'gudang_id');
}

}

File: Mobile TSTH2/inventory_tsth2/lib/Model/barangModel.dart
import 'package:inventory_tsth2/Model/gudangModel.dart';
import 'package:inventory_tsth2/Model/jenisbarangModel.dart';
import 'package:inventory_tsth2/Model/satuanModel.dart';

class BarangModel {
  int barangId;
  String barangKode;
  String barangNama;
  String barangHarga;
  String barangStok;
  String barangGambar;
  String barcodeUrl;
  JenisBarangModel? jenisBarang; 
  SatuanModel? satuan; 
  GudangModel? gudang; 

  BarangModel({
    required this.barangId,
    required this.barangKode,
    required this.barangNama,
    required this.barangHarga,
    required this.barangStok,
    required this.barangGambar,
    required this.barcodeUrl,
    this.jenisBarang, 
    this.satuan, 
    this.gudang, 
  });

  factory BarangModel.fromJson(Map<String, dynamic> json) {
    return BarangModel(
      barangId: json['barang_id'] ?? 0,
      barangKode: json['barang_kode'] ?? '',
      barangNama: json['barang_nama'] ?? '',
      barangHarga: json['barang_harga'] ?? '0',
      barangStok: json['barang_stok'] ?? '0',
      barangGambar: json['barang_gambar'] ?? '',
      barcodeUrl: json['barcodeUrl'] ?? '',
      jenisBarang: json['jenis_barang'] != null
          ? JenisBarangModel.fromJson(json['jenis_barang'])
          : null, // Bisa null
      satuan: json['satuan'] != null ? SatuanModel.fromJson(json['satuan']) : null,
      gudang: json['gudang'] != null ? GudangModel.fromJson(json['gudang']) : null,
    );
  }
}
